inception/inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasDocumentIndex.java

```diff
@@ -396,6 +402,221 @@ public long numberOfQueryResults(SearchQueryRequest aRequest)
+
+            for (int i = 0; i < reader.maxDoc(); i++) {
+                String sourceID = reader.document(i).get(FIELD_SOURCE_DOCUMENT_ID);
+                String annotationID = reader.document(i).get(FIELD_ANNOTATION_DOCUMENT_ID);
+                if (Long.valueOf(annotationID) != -1L) {
+                    if (reader.document(i).get(FIELD_USER)
+                            .equals(aStatisticRequest.getUser().getUsername())) {
+                        fullDocSet.add(i);
+                    }
+                }
+                // source document without annotation layer? then user is not relevant
+                else if (!annotatableDocuments.containsKey(Long.valueOf(sourceID))) {
+                    fullDocSet.add(i);
+                }
+            }
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+        }
+        finally {
+            if (searcher != null) {
+                // Releasing and setting to null per recommendation in JavaDoc of
+                // release(searcher)
+                // method
+                // getSearcherManager().release(searcher);
+                getSearcherManager().release(searcher);
+                searcher = null;
+            }
+        }
+        return fullDocSet;
+    }
+
+    public MtasSpanQuery parseQuery(String query) throws ExecutionException, Exception
+    {
+        final MtasSpanQuery mtasSpanQuery;
+        try {
+            String modifiedQuery = preprocessQuery(query);
+            MtasSpanQuery mtasSpanQuery1;
+            try (Reader queryReader = new StringReader(modifiedQuery)) {
+                MtasCQLParser parser = new MtasCQLParser(queryReader);
+                mtasSpanQuery1 = parser.parse(FIELD_CONTENT, DEFAULT_PREFIX, null, null, null);
+            }
+            mtasSpanQuery = mtasSpanQuery1;
+        }
+        catch (ParseException | Error e) {
+            // The exceptions thrown by the MTAS CQL Parser are inheriting from
+            // java.lang.Error...
+            throw new ExecutionException("Unable to parse query [" + query + "]", e);
+        }
+        return mtasSpanQuery;
+    }
+
+    @Override
+    public Map<String, Double> getLayerStatistics(StatisticRequest aStatisticRequest,
+            String aFeatureQuery, ArrayList<Integer> fullDocSet, boolean perSentence)
+        throws IOException, ExecutionException
+    {
+        IndexSearcher searcher = null;
+        Map<String, Object> resultsMap = null;
+        Map<String, Double> resultsMapDouble = new HashMap<String, Double>();
+        Map<Long, Long> annotatableDocuments = listAnnotatableDocuments(
+                aStatisticRequest.getProject(), aStatisticRequest.getUser());
+        try {
+            searcher = getSearcherManager().acquire();
+            IndexReader reader = searcher.getIndexReader();
+
+            // what does this parameter do?
+            ArrayList<Integer> fullDocList = new ArrayList<Integer>();
```

Comment:
List = ArrayList

Reply:
Then I need to do this cast because the mtas method requires ArrayList:
![grafik](https://user-images.githubusercontent.com/61779598/132345188-5652160f-0234-4376-a216-8640da85967f.png)